### PR TITLE
DeepLC integration

### PR DESCRIPTION
**Pull request by RalfG**
Integrate DeepLC to further improve PSM rescoring.

- [x] Continue restructuring code
    - [x] Move tests out of package directory
    - [x] Move code in __init__ to __main__, config_parser, runner
    - [x] Fix publish workflow and test file paths
- [x] New module peptide_record to handle PEPREC files (should be moved to MS²PIP)
- [x] Integrate DeepLC
- [x] Propagate required columns to peprec (`psm_score` and `observed_retention_time`)
    - [x] MaxQuant pipeline
    - [x] PeptideShaker pipeline
    - [x] X!Tandem pipeline
    - [ ] MS-GF+ pipeline
- [x] Update documentation

### CODE REVIEW

File: ms2rescore/retention_time.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+class RetentionTimeIntegration:
+    """Retention time integration for MS²ReScore, using DeepLC."""
+
+    def __init__(
+        self,
+        peprec_path: str,
+        feature_path: str,
+        higher_psm_score_better: bool = True,
+        num_calibration_psms: int = 500,
+        num_cpu: Union[int, None] = None,
```
Comment (paretje) [inline ms2rescore/retention_time.py:22]: `Optional[int]` is more readable.

File: ms2rescore/retention_time.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            n_jobs=self.num_cpu, cnn_model=True, verbose=False
+        )
+        self.peprec = None
+        self.feature_df = None
+
+        if self.peprec_path:
+            self.peprec = PeptideRecord(path=self.peprec_path)
+
+    @property
+    def calibration_df(self):
```
Comment (paretje) [inline ms2rescore/retention_time.py:72]: `calibration_data` might be clearer

File: ms2rescore/retention_time.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        ascending = not self.higher_psm_score_better
+        return (
+            self.peprec.df.sort_values(["psm_score"], ascending=ascending)
+            .sample(self.num_calibration_psms)
+            .copy()
+            .rename(columns={"observed_retention_time": "tr", "peptide": "seq",})
+        )
+
+    @property
+    def prediction_df(self):
```
Comment (paretje) [inline ms2rescore/retention_time.py:90]: `prediction_data`? `_df` seems somewhat cryptic.

File: ms2rescore/retention_time.py (hunk 1 above)
Comment (RalfG) [reply ms2rescore/retention_time.py:22]: Cool! Didn't know about this notation. I haven't really found a good resource on type hinting yet...

Comment (paretje) [reply ms2rescore/retention_time.py:22]: The PEP might be a good one: https://www.python.org/dev/peps/pep-0484/

**Pull request merged**